library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt

```diff
@@ -28,60 +28,29 @@ import kotlin.jvm.Throws
- * results.
- * @param cacheDirectoryProvider Provider of [File] where Chucker will save temporary responses
- * before processing them.
- * @param alwaysReadResponseBody If set to `true` Chucker will read full content of response
- * bodies even in case of parsing errors or closing the response body without reading it.
- * @param headersToRedact a [Set] of headers you want to redact. They will be replaced
- * with a `**` in the Chucker UI.
  */
-public class ChuckerInterceptor internal constructor(
-    private val context: Context,
-    private val collector: ChuckerCollector = ChuckerCollector(context),
-    private val maxContentLength: Long = 250000L,
-    private val cacheDirectoryProvider: CacheDirectoryProvider,
-    private val alwaysReadResponseBody: Boolean = false,
-    headersToRedact: Set<String> = emptySet(),
+public class ChuckerInterceptor private constructor(
+    builder: Builder,
 ) : Interceptor {
 
     /**
      * An OkHttp Interceptor which persists and displays HTTP activity
      * in your application for later inspection.
      *
+     * This constructor  is a shorthand for `ChuckerInterceptor.Builder(context).build()`.
+     *
      * @param context An Android [Context]
-     * @param collector A [ChuckerCollector] to customize data retention
-     * @param maxContentLength The maximum length for request and response content
-     * before their truncation. Warning: setting this value too high may cause unexpected
-     * results.
-     * @param alwaysReadResponseBody If set to `true` Chucker will read full content of response
-     * bodies even in case of parsing errors or closing the response body without reading it.
-     * @param headersToRedact a [Set] of headers you want to redact. They will be replaced
-     * with a `**` in the Chucker UI.
+     * @see ChuckerInterceptor.Builder
      */
-    @JvmOverloads
-    public constructor(
-        context: Context,
-        collector: ChuckerCollector = ChuckerCollector(context),
-        maxContentLength: Long = 250000L,
-        headersToRedact: Set<String> = emptySet(),
-        alwaysReadResponseBody: Boolean = false,
-    ) : this(
-        context = context,
-        collector = collector,
-        maxContentLength = maxContentLength,
-        cacheDirectoryProvider = { context.cacheDir },
-        alwaysReadResponseBody = alwaysReadResponseBody,
-        headersToRedact = headersToRedact,
-    )
-
-    private val io: IOUtils = IOUtils(context)
-    private val headersToRedact: MutableSet<String> = headersToRedact.toMutableSet()
+    public constructor(context: Context) : this(Builder(context))
+
+    private val context = builder.context
+    private val collector = builder.collector ?: ChuckerCollector(context)
+    private val maxContentLength = builder.maxContentLength
+    private val cacheDirectoryProvider = builder.cacheDirectoryProvider ?: CacheDirectoryProvider { context.filesDir }
```

Comment:
This is the inconvenience that I mentioned. Default fields are declared here instead of builder. It is only due to test purposes. Instantiating `ChuckerCollector` would require a lot of mocking of `Context` that I don't want to be botheredwith.

On the other hand it is nice to not create stuff eagerly in the builder.